api/src/test/java/marquez/DatasetIntegrationTest.java

```diff
@@ -495,4 +452,109 @@ public void testApp_getDatasetContainsColumnLineage() {
+  }
+
+  @Test
+  public void testApp_doesNotShowDeletedDatasetAfterUndeleteNamespace() throws IOException {
+    String namespaceName = "namespace";
+    String name = "table";
+
+    LineageEvent firstEvent =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job(namespaceName, "job_name", null),
+            List.of(
+                new LineageEvent.Dataset(namespaceName, name, LineageTestUtils.newDatasetFacet())),
+            Collections.emptyList(),
+            "the_producer");
+
+    LineageEvent secondEvent =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job(namespaceName, "second_job_name", null),
+            List.of(
+                new LineageEvent.Dataset(
+                    namespaceName, name + "2", LineageTestUtils.newDatasetFacet())),
+            Collections.emptyList(),
+            "the_producer");
+
+    CompletableFuture<Integer> resp = sendEvent(firstEvent);
+    assertThat(resp.join()).isEqualTo(201);
+
+    resp = sendEvent(secondEvent);
+    assertThat(resp.join()).isEqualTo(201);
+
+    List<Dataset> datasets = client.listDatasets(namespaceName);
+    assertThat(datasets).hasSize(2);
+
+    client.deleteNamespace(namespaceName);
+
+    List<Namespace> namespaces = client.listNamespaces();
+    assertThat(namespaces)
+        .anySatisfy(
+            namespace -> {
+              assertThat(namespace.getIsHidden()).isTrue();
+              assertThat(namespace.getName()).isEqualTo(namespaceName);
+            });
+
+    datasets = client.listDatasets(namespaceName);
+    assertThat(datasets).hasSize(0);
+
+    List<Job> jobs = client.listJobs(namespaceName);
+    assertThat(jobs).hasSize(0);
+
+    LineageEvent thirdEvent =
+        new LineageEvent(
+            "COMPLETE",
+            Instant.now().atZone(ZoneId.systemDefault()),
+            new LineageEvent.Run(UUID.randomUUID().toString(), null),
+            new LineageEvent.Job(namespaceName, "job_name", null),
+            List.of(
+                new LineageEvent.Dataset(namespaceName, name, LineageTestUtils.newDatasetFacet())),
+            Collections.emptyList(),
+            "the_producer");
+
+    resp = sendEvent(thirdEvent);
+    assertThat(resp.join()).isEqualTo(201);
+
+    namespaces = client.listNamespaces();
+    assertThat(namespaces)
+        .anySatisfy(
+            namespace -> {
+              assertThat(namespace.getIsHidden()).isFalse();
+              assertThat(namespace.getName()).isEqualTo(namespaceName);
+            });
+
+    datasets = client.listDatasets(namespaceName);
+    assertThat(datasets).hasSize(1);
+
+    jobs = client.listJobs(namespaceName);
+    assertThat(jobs).hasSize(1);
+  }
```

Comment:
Could we add some `undelete` scenario?

Reply:
The test does check the number of datasets after the namespace is undeleted. But, looks like that's a new dataset added to the namespace and not the original datasets?

Reply:
Yes, the idea here is not to automatically undelete everything in the namespace.